src/lib/comms/sol-lwm2m.c

```diff
@@ -3014,9 +3014,106 @@ clear_bootstrap_ctx(struct sol_lwm2m_client *client)
     }
 }
 
+static bool
+is_authorized(struct sol_lwm2m_client *client,
```

Comment:
I suggest that you change this signature and return an int. By doing so we can know if an error happens during the read process.
So basically you can do something like:
If is_authrorized returns < 0 == Error
If is_authorized returns 0 == Not authorized.
If is_authorized returns > 0 == Authorized